lib/tasks/demo.rake

```diff
@@ -869,4 +901,64 @@ namespace :demo do
   def create_fake_timestamp(from, to)
     Faker::Time.between_dates(from: from, to: to >= Date.today ? Time.now : to, period: :all)
   end
+
+  # Duplicate patient and all nested relations and change last name
+  def deep_duplicate_patient(patient, responder_id = nil)
+    new_pat = patient.dup
+    new_pat.responder_id = responder_id unless responder_id.nil?
+    new_pat.last_name = "#{Faker::Name.last_name}#{rand(10)}#{rand(10)}"
+    new_pat.save
+    new_pat.update(responder_id: new_pat.id) if responder_id.nil?
+    patient.dependents.each do |p|
+      if p.id != p.responder_id
+         deep_duplicate_patient(p, new_pat.id)
+      end
+    end
+    patient.assessments.each do |a| 
+        newa = a.dup
+        newa.created_at = a.created_at
+        rep_condition = a.reported_condition
+        newr = rep_condition.dup
+        newr.save
+        symptoms = []
+        a.reported_condition.symptoms.each do |s|
+            news = s.dup
+            news.condition_id = newr.id
+            symptoms << news
+        end
+        Symptom.import symptoms
+        newa.patient_id = new_pat.id
+        newa.save
+        newr.update(assessment_id: newa.id)
+    end
+
+    histories = []
+    patient.histories.each do |h| 
+        newh = h.dup
+        newh.created_at = h.created_at
```

Comment:
is there a reason for not copying over `created_at` for other tables like transfers and close_contacts? what about `updated_at`?